image/storage/storage_dest.go

```diff
@@ -1586,8 +1594,13 @@ func (s *storageImageDestination) CommitWithOptions(ctx context.Context, options
 		// sizes (tracked in the metadata) which might have already
 		// been present with new values, when ideally we'd find a way
 		// to merge them since they all apply to the same image
+		// Create a digest function that uses the configured algorithm and handles schema1 manifests properly
+		digestFunc := func(data []byte) (digest.Digest, error) {
+			// Use manifest.Digest to handle schema1 signature stripping properly
+			return manifest.Digest(data)
+		}
```

Comment:
This is 2 new comments and 5 new lines to do exactly what the code did previously.

We _might_ need to parametrize this, and see elsewhere about `manifest.Digest`, but this is not helping.